app/com/arpnetworking/metrics/portal/alerts/impl/DatabaseAlertRepository.java

```diff
@@ -0,0 +1,376 @@
+                    .setPeriod(Period.seconds(ebeanAlert.getPeriod()))
+                    .setService(ebeanAlert.getService())
+                    .setStatistic(ebeanAlert.getStatistic())
+                    .setValue(new DefaultQuantity.Builder()
+                            .setValue(ebeanAlert.getQuantityValue())
+                            .setUnit(ebeanAlert.getQuantityUnit())
+                            .build())
+                    .setExtensions(mergeExtensions(ebeanAlert.getNagiosExtension()))
+                    .build();
+            values.add(result);
+        });
+
+        // Transform the results
+        return new DefaultQueryResult<>(values, pagedAlerts.getTotalRowCount(), etag);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getAlertCount() {
+        assertIsOpen();
+        return Ebean.find(models.ebean.Alert.class)
+                .findRowCount();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void addOrUpdateAlert(final Alert alert) {
+        assertIsOpen();
+        LOGGER.debug()
+                .setMessage("Upserting alert")
+                .addData("alert", alert)
+                .log();
+
+        final Transaction transaction = Ebean.beginTransaction();
+
+        try {
+            models.ebean.Alert ebeanAlert = Ebean.find(models.ebean.Alert.class)
+                    .where()
+                    .eq("uuid", alert.getId())
+                    .findUnique();
+            boolean created = false;
+            if (ebeanAlert == null) {
+                ebeanAlert = new models.ebean.Alert();
+                created = true;
+            }
+
+            ebeanAlert.setCluster(alert.getCluster());
+            ebeanAlert.setUuid(alert.getId());
+            ebeanAlert.setMetric(alert.getMetric());
+            ebeanAlert.setContext(alert.getContext());
+            ebeanAlert.setNagiosExtension(getNagiosExtensionFromMap(alert.getExtensions()));
+            ebeanAlert.setName(alert.getName());
+            ebeanAlert.setOperator(alert.getOperator());
+            ebeanAlert.setPeriod(alert.getPeriod().toStandardSeconds().getSeconds());
+            ebeanAlert.setQuantityValue(alert.getValue().getValue());
+            ebeanAlert.setQuantityUnit(alert.getValue().getUnit().isPresent() ? alert.getValue().getUnit().get() : null);
+            ebeanAlert.setStatistic(alert.getStatistic());
+            ebeanAlert.setService(alert.getService());
+            _alertQueryGenerator.saveAlert(ebeanAlert);
+            transaction.commit();
+
+            LOGGER.info()
+                    .setMessage("Upserted alert")
+                    .addData("alert", alert)
+                    .addData("isCreated", created)
+                    .log();
+        } finally {
+            transaction.end();
+        }
+    }
+
+    //NOTE: Only to be used to clear the tables for testing
+    /**
+     * Package private.
+     */
+    void deleteAll() {
```

Comment:
I really don't think this is a good idea.  Even if it's for testing, use of this function means that tests can't run in parallel.